### DISCUSSION TITLE
Samsung Workaround also for Oppo?

### DISCUSSION THREAD

**Discussion by Miquest** [NightscoutFoundation/xDrip#2732] [Ideas]
Dear developers,
I've been following the discussion regarding the Oneplus workaround for the Dexcom G6 from about a month ago.
I have an Oppo smartphone (Android 13) and the exact same problem. Could you try the buggy Samsung workaround for Oppo smartphones as well?
I could test it directly and quickly give feedback on the workaround on Oppo smartphones. That would help me a lot!

Kind regards,
Miquest

**Comment by Navid200** [comment]
Why do you think it's the same problem?

**Reply by Miquest** [reply]
I also have very high jitter as previously described. About 5 minutes for me, so my smartphone is not able to receive any value from the Dexcom G6.
Why am I so sure it's the same problem? Well, ColorOS (by Oppo) and OxygenOS (by Oneplus) are almost identical architecturally. So I think we can solve this problem in the same way. Unfortunately I can't share any screenshots or logs as I switched to the BYODA shortly after I had this problem.

**Reply by Navid200** [reply]
I would like to add this workaround for all phones.
I cannot promise it will be merged.
But, it will be my next PR.  I hope to open the PR in a few days up to a week max from now.

**Reply by Miquest** [reply]
Is there a Branch on GitHub where I can build your app version by myself?
Or do you have your own repo where I can download your modifications for the app (such as "Very Low" and "Very High" alert)?

**Reply by Miquest** [reply]
Thank you that you're working in it!

**Reply by Navid200** [reply]
Even though xDrip is open source, it is not my intent to release my personal versions.  That would divide the community.  
I personally get a rewarding satisfaction being able to serve the community.  Dividing it goes against that.

My very high and very low PR is to add those to the statistics to include on the distribution page.
If what you need is a very high or very low alert, you can already create those with xDrip as is.  You can create 10s of alerts if you desire.

If you are interested in a PR, you can state that in the PR.  But, if you do that, please make it factual.  Not emotional.

xDrip must be maintained.  Any addition could open the door for more bugs.  We cannot just keep adding features (even if they are my PRs).  Every addition must have a justified reason.
The person who has to make the judgement has a very difficult job.  No matter what decision he makes, someone is likely to be pissed at him for that decision.
Please be patient.

**Reply by Miquest** [reply]
Okay, thank you for claryfiing.
I was just wondering where the PR's you do come from. That is why I asked. I thought I could download and install an unreleased build from this branch.
In no case do I want the community to be divided, it's important that there is such a thing! 
So I'll wait a little longer and continue to use the BYODA until the workaround has found its way into xDrip Plus.

**Reply by Navid200** [reply]
Give it a try now: https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.03.23

**Reply by Miquest** [reply]
I downloaded the new version already, I activated email notifications for xDrip+ repository. 
It's working perfectly until now, thank you very much!
I just had to force stop of xDrip in my phones settings and to restart the app then. 
Previously, the app didn't recognize that my phone needed the workaround, but it could also be that it was because of the switch from 640G/Eversense to G5/G6/ONE as the data source.

Thank you very very much for realizing that!
